addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusSymoInverterHandler.java

```diff
@@ -1,230 +1,246 @@
+        switch (fieldName) {
+            case FroniusBindingConstants.InverterDataChannelDayEnergy:
+                ValueUnit day = inverterRealtimeResponse.getBody().getData().getDayEnergy();
+                day.setUnit("kWh");
+                return day;
+            case FroniusBindingConstants.InverterDataChannelPac:
+                return inverterRealtimeResponse.getBody().getData().getPac();
+            case FroniusBindingConstants.InverterDataChannelTotal:
+                ValueUnit total = inverterRealtimeResponse.getBody().getData().getTotalEnergy();
+                total.setUnit("MWh");
+                return total;
+            case FroniusBindingConstants.InverterDataChannelYear:
+                ValueUnit year = inverterRealtimeResponse.getBody().getData().getYearEnergy();
+                year.setUnit("MWh");
+                return year;
+            case FroniusBindingConstants.InverterDataChannelFac:
+                return inverterRealtimeResponse.getBody().getData().getFac();
+            case FroniusBindingConstants.InverterDataChannelIac:
+                return inverterRealtimeResponse.getBody().getData().getIac();
+            case FroniusBindingConstants.InverterDataChannelIdc:
+                return inverterRealtimeResponse.getBody().getData().getIdc();
+            case FroniusBindingConstants.InverterDataChannelUac:
+                return inverterRealtimeResponse.getBody().getData().getUac();
+            case FroniusBindingConstants.InverterDataChannelUdc:
+                return inverterRealtimeResponse.getBody().getData().getUdc();
+        }
+        if (powerFlowResponse == null || powerFlowResponse.getBody() == null
+                || powerFlowResponse.getBody().getData() == null
+                || powerFlowResponse.getBody().getData().getSite() == null) {
+            return null;
+        }
+        switch (fieldName) {
+            case FroniusBindingConstants.PowerFlowpGrid:
+                return powerFlowResponse.getBody().getData().getSite().getPgrid();
+            case FroniusBindingConstants.PowerFlowpLoad:
+                return powerFlowResponse.getBody().getData().getSite().getPload();
+            case FroniusBindingConstants.PowerFlowpAkku:
+                return powerFlowResponse.getBody().getData().getSite().getPakku();
+        }
+
+        return null;
+    }
+
+    /**
+     * Get new data
+     */
+    private void updateData(FroniusBridgeConfiguration bridgeConfiguration, FroniusBaseDeviceConfiguration config) {
+        inverterRealtimeResponse = getRealtimeData(bridgeConfiguration.hostname, config.deviceId);
+        powerFlowResponse = getPowerFlowRealtime(bridgeConfiguration.hostname);
+    }
+
+    /**
+     * Make the PowerFlowRealtimeDataRequest
+     *
+     * @param ip address of the device
+     * @return {PowerFlowRealtimeResponse} the object representation of the json response
+     */
+    private PowerFlowRealtimeResponse getPowerFlowRealtime(String ip) {

```

Comment:
The methods `getPowerFlowRealtime` and `getRealtimeData` are almost the same you could extract the common part into a single method and parameterize that method with the difference.